src/app/public/modules/country-field/country-field.component.spec.ts

```diff
@@ -0,0 +1,380 @@
+} from '@skyux-sdk/testing';
+
+import {
+  SkyCountryFieldModule
+} from './country-field.module';
+
+import {
+  CountryFieldTestComponent
+} from './fixtures/country-field.component.fixture';
+
+import {
+  CountryFieldReactiveTestComponent
+} from './fixtures/country-field-reactive.component.fixture';
+
+describe('Country Field Component', () => {
+
+  function blurInput(fixture: ComponentFixture<any>): void {
+    SkyAppTestUtility.fireDomEvent(getInputElement(), 'blur');
+    fixture.detectChanges();
+    tick();
+  }
+
+  function enterSearch(newValue: string, fixture: ComponentFixture<any>): void {
+    const inputElement = getInputElement();
+    inputElement.value = newValue;
+
+    SkyAppTestUtility.fireDomEvent(inputElement, 'keyup');
+    fixture.detectChanges();
+    tick();
+    fixture.detectChanges();
+    tick();
+  }
+
+  function getAutocompleteElement(): HTMLElement {
+    return document.querySelector('sky-autocomplete') as HTMLElement;
+  }
+
+  function getInputElement(): HTMLTextAreaElement {
+    return document.querySelector('textarea') as HTMLTextAreaElement;
+  }
+
+  function searchAndSelect(newValue: string, index: number, fixture: ComponentFixture<any>): void {
+    const inputElement = getInputElement();
+
+    enterSearch(newValue, fixture);
+    const searchResults = getAutocompleteElement().querySelectorAll('.sky-dropdown-item') as NodeListOf<HTMLElement>;
+
+    // Note: the ordering of these events is important!
+    SkyAppTestUtility.fireDomEvent(inputElement, 'change');
+    searchResults[index].querySelector('button').click();
+    blurInput(fixture);
+  }
+
+  function searchAndGetResults(newValue: string, fixture: ComponentFixture<any>): NodeListOf<HTMLElement> {
+    enterSearch(newValue, fixture);
+    return getAutocompleteElement().querySelectorAll('.sky-dropdown-item') as NodeListOf<HTMLElement>;
+  }
+
+  describe('template form', () => {
+
+    let fixture: ComponentFixture<CountryFieldTestComponent>;
+    let component: CountryFieldTestComponent;
+    let nativeElement: HTMLElement;
+
+    beforeEach(() => {
+      TestBed.configureTestingModule({
+        declarations: [
+          CountryFieldTestComponent
+        ],
+        imports: [
+          FormsModule,
+          SkyCountryFieldModule
+        ]
+      });
+
+      fixture = TestBed.createComponent(CountryFieldTestComponent);
+      nativeElement = fixture.nativeElement as HTMLElement;
+      component = fixture.componentInstance;
+    });
+
+    describe('initialization', () => {
+
+      it('should initialize with a set country', fakeAsync(() => {
+        component.modelValue = {
+          name: 'United States',
+          iso2: 'us'
+        };
+        fixture.detectChanges();
+        tick();
+        fixture.detectChanges();
+
+        expect(nativeElement.querySelector('textarea').value).toBe('United States');
+        expect(nativeElement.querySelector('.sky-country-field-flag')).toHaveCssClass('iti-flag');
```

Comment:
I'm seeing a lot of tests for `.toHaveCssClass('iti-flag')`, and no tests for the opposite. In your dom, you have an ngIf that toggles this class on/off - should we write two tests that cover these two scenarios?

Reply:
The opposite here is `expect(nativeElement.querySelector('.sky-country-field-flag')).toBeNull();` which we have tests for. There is never a case where that isn't null and the flag isn't there.

Reply:
Works for me, thanks.